internal/rule/exists.go

```diff
@@ -0,0 +1,144 @@
+package rule
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type Exists struct {
+	name      string
+	exclusive bool
+	min       int16
```

Comment:
You are using a limited int, maybe for memory considerate. I'm unsure.

I would have used int.

But if you are looking for performance or memory footprint maybe you could use uint16


```suggestion
	min       uint16
```